File: 9/9.py
#
# File: 9.py
# Description: http://projecteuler.net/problem=9
#

SUM=1000

print reduce(lambda x, y: x*y, 
             [(i, j - i, SUM - j) 
              for i in range(1, SUM) 
              for j in range(i + 1, SUM) 
              if ((SUM - j)**2 == i**2 + (j - i)**2)][0])

File: 9/9.c
/*
 * File: 9.c
 * Description: http://projecteuler.net/problem=9
 */

#include<stdio.h>

int main() {
    int i, j, a, b, c;
    for(i = 1; i < 1000; i++)
        for(j = i + 1; j < 1000; j++) {
            a = i;
            b = j - i;
            c = 1000 - (b + a);
            if (c*c == a*a + b*b) {
                printf("%d\n", a*b*c);
                return 0;
            }
        }
}
